pkg/resources/management.cattle.io/v3/globalrole/validator.go

```diff
@@ -121,49 +120,42 @@ func (a *admitter) Admit(request *admission.Request) (*admissionv1.AdmissionResp
 		return nil, err
 	}
 
+	// check if the user has the escalate verb
+	hasEscalate, escErr := auth.RequestUserHasVerb(request, gvr, a.sar, escalateVerb, newGR.Name, "")
+	if escErr != nil {
+		logrus.Warnf("Failed to check for the 'escalate' verb on GlobalRoles: %v", escErr)
+	}
+
+	// if we have the escalate verb, no need to check global/cluster/namespaced permissions
+	if hasEscalate {
+		return admission.ResponseAllowed(), nil
+	}
```

Comment:
I am aware of this and am doing some investigation to see the performance impact. My hunch is you are correct since the SAR check is an API call vs just checking the cache for the rules and is therefore much slower and we should avoid it.

My only thought is that given there could be many rules to look at, there may be some threshold where it would be better to check the verb first. I am spending some time to investigate this, but in the mean time will address the rest of the comments.